cpp/src/Ice/ConnectionI.cpp

```diff
@@ -2321,17 +2321,22 @@ Ice::ConnectionI::inactivityCheck() noexcept
 {
     // Called by the InactivityTimerTask.
     std::lock_guard lock(_mutex);
-    if (_state == StateActive && _inactivityTimerTaskScheduled && // it's false if some other thread canceled the timer
-                                                                  // while we were waiting for _mutex
-        !_timer->isScheduled(_inactivityTimerTask)) // make sure this timer task was not rescheduled for later while
-                                                    // we were waiting for _mutex (unlikely but may as well check)
-    {
-        setState(
-            StateClosing,
-            make_exception_ptr(ConnectionClosedException{
-                __FILE__,
-                __LINE__,
-                "connection closed because it remained inactive for longer than the inactivity timeout"}));
+
+    // Make sure this timer task was not rescheduled for later while we were waiting for _mutex.
+    if (_inactivityTimerTaskScheduled && !_timer->isScheduled(_inactivityTimerTask))
+    {
+        // Clear flag - the task is no longer scheduled.
+        _inactivityTimerTaskScheduled = false;
```

Comment:
That's the difference with the previous version.  

We must set this flag to false in case the inactivity timer task runs while we're in the holding state, and then the application switches back to active but no activity takes place. We want to inactivity timeout to be rescheduled in this case.